plasmapy/particles/particle_class.py

```diff
@@ -2042,48 +2049,79 @@ class CustomParticle(AbstractPhysicalParticle):
-    will be set to |nan| in the appropriate units.
+    ~plasmapy.particles.particle_collections.ParticleList
 
     Examples
     --------
     >>> from astropy import units as u
     >>> from plasmapy.particles import CustomParticle
-    >>> custom_particle = CustomParticle(
-    ...     mass=1.2e-26 * u.kg,
-    ...     charge=9.2e-19 * u.C,
-    ... )
+
+    >>> custom_particle = CustomParticle(9.2e-19 * u.C, 1.2e-26 * u.kg, "φ")
     >>> custom_particle.mass
     <Quantity 1.2e-26 kg>
     >>> custom_particle.charge
     <Quantity 9.2e-19 C>
+    >>> custom_particle.symbol
+    'φ'
+
     >>> average_particle = CustomParticle(
     ...     mass=1.5e-26 * u.kg,
     ...     Z = -1.5,
-    ...     symbol="Ξ",
+    ...     symbol="X",
     ... )
-    >>> average_particle.mass
-    <Quantity 1.5e-26 kg>
     >>> average_particle.charge
     <Quantity -2.40326...e-19 C>
-    >>> average_particle.symbol
-    'Ξ'
     """
 
     def __init__(
         self,
-        mass: u.kg = None,
-        charge: u.C = None,
+        *quantities,
+        charge: Optional[u.C] = None,
+        mass: Optional[u.kg] = None,
         symbol: Optional[str] = None,
-        *,
         Z: Optional[Real] = None,
     ):
 
         # TODO py3.10 replace ifology with structural pattern matching
 
+        if quantities:
+            if isinstance(quantities[-1], str) and symbol is None:
+                symbol = quantities[-1]
+                quantities = list(quantities[:-1])
+
+            try:
+                physical_type_dict = _get_physical_type_dict(
+                    quantities,
+                    only_quantities=True,
+                    strict=True,
+                    allowed_physical_types={
+                        u.physical.mass,
+                        u.physical.electrical_charge,
+                    },
+                )
+            except (TypeError, ValueError) as exc:
+                raise InvalidParticleError(
+                    "Unable to create a CustomParticle based off of the "
+                    "following positional arguments, which should be a"
+                    f"Quantity representing electrical charge and/or a"
+                    f"Quantity representing mass: {quantities}."
+                ) from exc
+
+            if u.physical.electrical_charge in physical_type_dict:
+                if charge is not None:
+                    raise TypeError(
+                        "Cannot provide charge as both a positional and keyword "
+                        "argument."
+                    )
+                charge = physical_type_dict[u.physical.electrical_charge]
+
+            if u.physical.mass in physical_type_dict:
+                if mass is not None:
+                    raise TypeError(
+                        "Cannot provide mass as both a positional and keyword argument."
+                    )
+                mass = physical_type_dict[u.physical.mass]
+
```

Comment:
Since this whole block is pretty separate from other functionality, how about moving it to a separate semi-private constructor like this...
```python
@classmethod
def _from_positional(cls, args):
    ... # this whole block
```

and then calling that from here if `quantities` is non-empty?

Reply:
Agreed.  This would be cleaner by extracting a bunch of this into its own method.